src/api/fuel.ts

```diff
@@ -0,0 +1,548 @@
+  getKeys = async (): Promise<string[]> => this.user.getKeys();
+}
+
+// Auxiliar functions to validate with the user the use of the service
+interface Preference {
+  remember?: boolean;
+  approve?: boolean;
+}
+export default class GreymassFuelService {
+  static preferences: { [account: string]: Preference } = {};
+  static globals: Record<string, (s: string) => string> = null;
+  static save() {
+    localStorage.setItem(
+      'fuel_preferences',
+      JSON.stringify(GreymassFuelService.preferences)
+    );
+  }
+  static load() {
+    try {
+      GreymassFuelService.preferences = GreymassFuelService.preferences || {};
+      const str = localStorage.getItem('fuel_preferences');
+      if (str) {
+        GreymassFuelService.preferences = JSON.parse(str);
+      }
+    } catch (e) {
+      console.error('ERROR: ', e);
+    }
+  }
+
+  static setPreferences(account: string, p: Preference) {
+    GreymassFuelService.preferences[account] = {
+      ...GreymassFuelService.preferences[account],
+      ...p
+    };
+    if (GreymassFuelService.preferences[account].remember) {
+      GreymassFuelService.save();
+    }
+  }
+  static setGlobals(g: Record<string, (s: string) => string>) {
+    GreymassFuelService.globals = g;
+  }
+}
+
+async function confirmWithUser(user: User) {
+  const username = await user.getAccountName();
+  let mymodel: string[] = [];
+  mymodel = [];
+
+  // TODO: when localization is active we can do take from here
+  // const $t:(s:string) => string = GreymassFuelService.globals['$t'];
+  // console.log(typeof $t);
+
+  return new Promise<void>((resolve, reject) => {
+    // Try and see if the user already answer (remembered)
+    if (
+      GreymassFuelService.preferences[username] &&
+      GreymassFuelService.preferences[username].remember
+    ) {
+      // ok, the user did. What's the answer?
+      if (GreymassFuelService.preferences[username].approve) {
+        resolve();
+      } else {
+        reject();
+      }
+      return;
+    }
+
+    const handler = function (approve: boolean) {
+      GreymassFuelService.setPreferences(username, { approve });
+      if (approve) resolve();
+      else reject();
+    };
+
+    // this are the normal texts for random wallet.
+    let cancel: string | boolean = 'Reject';
+    let ok = 'Approve';
+    let message =
+      "It seems you don't have enought resources to pay for your next transaction. " +
+      "But, don't worry. You can continue using Greymass Fuel services for free. Do you accept to use this service?";
+
+    // If the wallet is Greymass Anchor is not possible to avoid Fuel service (it is incorporated)
+    try {
+      if (user instanceof AnchorUser) {
+        cancel = false;
+        ok = 'Continue';
+        message =
+          "It seems you don't have enought resources to pay for your next transaction. " +
+          "But, don't worry. Graymass Anchor wallet incorporates Fuel service for you to connitue for free.";
```

Comment:
@Kersten-TCD, this is the text for the Anchor wallet case which does not give the option to skip the Fuel service.

Reply:
Anchor (via anchor-link's SDKs) should be able to handle all of the messaging for you, if you just want to skip any messaging on your end. The messaging is probably useful for non-Anchor clients though!